### ISSUE TITLE
[dynamo] torch._dynamo.exc.Unsupported: call_function BuiltinVariable(setattr) [TensorVariable(), ConstantVariable(str), ConstantVariable(bool)] {}

### ISSUE THREAD

**Issue by ezyang** [pytorch/pytorch#109505]
### 🐛 Describe the bug

Occurs when you do

```
foo.requires_grad = True
```

in torch.compile'd region.

Occurred while implementing torch.tensor polyfill.

### Versions

main

cc @msaroufim @wconstab @bdhirsh @anijain2305 @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng

**Comment by jansel** [comment]
Should we be wrapping this in an implicit autograd op?

**Comment by ezyang** [comment]
Yes, this needs to get to AOTAutograd in the end

**Comment by ezyang** [comment]
@bdhirsh we should graph break on this case

**Comment by bdhirsh** [comment]
Related: https://github.com/pytorch/pytorch/issues/107861

**Comment by jon-chuang** [comment]
Just curious, `requires_grad` flag is only in effect at the time when we call `loss.backward()`, correct? Meaning that we do not care about the flag any time before and after when running the graph? @bdhirsh

**Comment by ezyang** [comment]
We are obligated to respect it for AOTAutograd which runs backward, and we have to make sure the mutation is actually applied by the compiled code

**Comment by jon-chuang** [comment]
> and we have to make sure the mutation is actually applied by the compiled code

Hmm, I guess we have to graph break because:
- even though we can trace the mutations in FakeTensor, we still need to apply the mutations to the actual tensors which we call the compiled function on? However, for intermediate results created by the graph, these should already know whether they require_grad as propagated by the `FakeTensor`, correct?

Btw, why is context manager different? Is it because for `with torch.no_grad` prunes away `ops` i.e. nodes, while `.requires_grad(False)` prunes away tensors i.e. edges? Likewise, `with torch.enable_grad` can only create new intermediate leaves, while `torch.requires_grad` can create user-owned leaves?
